jazz_asset-event-handler/index.js

```diff
@@ -0,0 +1,360 @@
+            "username": configData.SERVICE_USER,
+            "password": configData.TOKEN_CREDS
+        },
+        rejectUnauthorized: false,
+        transform: function (body, response, resolveWithFullResponse) {
+            return response;
+        }
+    };
+};
+
+var getAuthResponse = function (result) {
+    logger.info("result.statusCode" + result.statusCode)
+    return new Promise((resolve, reject) => {
+        if (result.statusCode === 200 && result.body && result.body.data) {
+            return resolve(result.body.data.token);
+        } else {
+            logger.error("getAuthResponse failed");
+            return reject(errorHandler.throwInternalServerError("Invalid token response from API"));
+        }
+    })
+}
+
+var processEvents = function (event, configData, authToken) {
+    return new Promise((resolve, reject) => {
+        var processEachEventPromises = [];
+        for (var i = 0; i < event.Records.length; i++) {
+            processEachEventPromises.push(processEachEvent(event.Records[i], configData, authToken));
+        }
+        Promise.all(processEachEventPromises)
+            .then((result) => {
+                return resolve(result);
+            })
+            .catch((error) => {
+                logger.error("processEvents failed" + JSON.stringify(error));
+                return reject(error);
+            });
+    });
+}
+
+var processEachEvent = function (record, configData, authToken) {
+    return new Promise((resolve, reject) => {
+        var sequenceNumber = record.kinesis.sequenceNumber;
+        var encodedPayload = record.kinesis.data;
+        var payload;
+        return checkForInterestedEvents(encodedPayload, sequenceNumber, configData)
+            .then(result => {
+                payload = result.payload;
+                if (result.interested_event) {
+                    return processItem(payload, configData, authToken);
+                } else {
+                    return new Promise((resolve, reject) => {
+                        resolve({ "message": "Not an interesting event" });
+                    });
+                }
+            })
+            .then(result => {
+                handleProcessedEvents(sequenceNumber, payload);
+                return resolve(result);
+            })
+            .catch(err => {
+                logger.error("processEachEvent failed for " + JSON.stringify(record));
+                handleFailedEvents(sequenceNumber, err.failure_message, payload, err.failure_code);
+                return reject(err);
+            });
+    });
+}
+
+var checkForInterestedEvents = function (encodedPayload, sequenceNumber, config) {
+    return new Promise((resolve, reject) => {
+        var kinesisPayload = JSON.parse(new Buffer(encodedPayload, 'base64').toString('ascii'));
+        if (kinesisPayload.Item.EVENT_TYPE && kinesisPayload.Item.EVENT_TYPE.S) {
+            if (_.includes(config.EVENTS.EVENT_TYPE, kinesisPayload.Item.EVENT_TYPE.S) &&
```

Comment:
think about uisng Array.prototype.includes instead of lodash here and in other places in your code.

Reply:
Let's address these changes later.

Reply:
What was agreed upon here? Lets try to close comments in the PR.

Reply:
Well i don't necessarily agree with that. Rather than making big radical changes we should at least start doing that on a new service(baby steps) that we are creating/porting. This is a better approach in ensuring we are going in right direction rather than taking it all and then testing for regression. The changes you have to do are miniscule. It will take less than 2 minutes to implement this change. I have already provided an example in my first comment.

Reply:
So then it wasn't agreed upon. PR was merged w/o sign off then?

Reply:
What was agreed upon was that Deepu did not have the time to address this at that point in time but had agreed to open a new PR soon to address this.